graviton/invariant.py

```diff
@@ -10,14 +11,47 @@
 )
 
 
+class PredicateKind(Enum):
+    Constant = "constant"
+    CaseMap = "case map"
+    ExactMatch = "exact match"
+    RangeMatch = "range match"
+    IdenticalPair = "identical"
+
+
 INVARIANT_TYPE = Union[
     PyTypes,
     Dict[Sequence[PyTypes], PyTypes],
     Callable[[PyTypes], PyTypes],
     Callable[[PyTypes], bool],
```

Comment:
minor question: seems according to `models.py`, `PyTypes = Union[bool, int, Sequence, str, bytes]`, which encapsulate the next type `Callable[[PyTypes], bool]`. How are they distinguishable to each other?

Reply:
This was a good call out. These type hints are purely informational, but they weren't correct. I've adjusted them after I had a closer look.

Reply:
(see the next commit)